src/api.ts

```diff
@@ -330,16 +334,34 @@ function createApi({
         'INSERT INTO Tileset (id, tilejson, format, upstreamTileUrls) ' +
           'VALUES (:id, :tilejson, :format, :upstreamTileUrls)'
       ).run({
-        id,
+        id: tilesetId,
         format: tilejson.format,
         tilejson: JSON.stringify(tilejson),
         upstreamTileUrls: JSON.stringify(tilejson.tiles),
       })
 
       const result = {
         ...tilejson,
-        tiles: [getTileUrl(id)],
-        id,
+        id: tilesetId,
+        tiles: [getTileUrl(tilesetId)],
+      }
+
+      // Create raster style if tileset format is a raster format
+      if (tilejson.format !== 'pbf') {
+        const rasterStyle = createRasterStyleFromTileset(result)
+
+        // TODO: Ideally could reuse createStyle here
+        db.prepare<{
+          id: string
+          sourceIdToTilesetId: string
+          stylejson: string
+        }>(
+          'INSERT INTO Style (id, sourceIdToTilesetId, stylejson) VALUES (:id, :sourceIdToTilesetId, :stylejson)'
+        ).run({
+          id: getStyleId(),
+          sourceIdToTilesetId: JSON.stringify({ [tilesetId]: tilesetId }),
```

Comment:
a little awkward but in this case, the lone source id is the same as corresponding the tileset id